src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -254,6 +257,26 @@ def app_post_deploy_hooks(self) -> Optional[List[ApplicationPostDeployHook]]:
     def debug_mode(self) -> bool:
         return self.na_project.debug_mode
 
+    @contextmanager
+    def use_warehouse(self, new_wh: Optional[str]):
+        """
+        Switches to a different warehouse for a while, then switches back.
+        This is a no-op if the requested warehouse is already active, or the requested warehouse is not available.
+        """
+        wh_result = self._execute_query(
+            f"select current_warehouse()", cursor_class=DictCursor
+        ).fetchone()
+        prev_wh = wh_result["CURRENT_WAREHOUSE()"]
+        is_different_wh = (new_wh is not None) and (new_wh.lower() != prev_wh.lower())
```

Comment:
do we need to pay any consideration for quoted/unquoted identifiers?

Reply:
I don't believe so. The counterpart `use_role` and the subcall to `use` also do not use it.

Reply:
I am just thinking about the usecase where you have 2 different warehouses with the same chars: 'Warehouse' and 'warehousE'. It's a very corner case, but if it's ignored in other places (like role), maybe we can leave it out of scope for this PR.

Reply:
Can we re-use an existing utility function--or extract it out from somewhere--to check if the two identifiers are equal? I think these are the "identifier as string in resultset" so the rules are slightly different. Also, does this code currently fail if `prev_wh is None`?

Reply:
+1, this comparison seems wrong, it needs to consider quoted vs unquoted IDs at least (not to mention the case sensitivity param, but I'm OK ignoring that)

Reply:
Added the built-in `to_identifier` utility in the new commit, up for review.